### PR TITLE
Remove Travis config and documentation

### PR DESCRIPTION
**Pull request by Scotchester**
We've been successfully using GitHub Actions for a month or so; time to stop running Travis builds for this project.

Also does a little bit of docs housekeeping for some things I noticed as I was removing mentions of Travis from the docs.

## Removals

- Travis configuration and related documentation

## Changes

- A few minor doc updates unrelated to Travis

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -39,13 +37,12 @@ Versions for back-end software including Django, Wagtail, Jinja, etc. are kept i
 https://github.com/cfpb/cfgov-refresh/tree/master/requirements
 
 - `base.txt`: shortcut for `django.txt` + `wagtail.txt` + `libraries.txt`
-- `deployment.txt`: requirements for deployment, includes `base.txt` and `postgres.txt` and a New Relic library which we don't install anywhere else.
+- `ci.txt`: specific requirements for the continuous integration environment. Should/could be moved to CI configuration files?
```
Comment (higs4281) [inline docs/index.md:40]: Did you want this question in the docs?

File: docs/index.md [hunk 2]
```diff
@@ -39,13 +37,12 @@ Versions for back-end software including Django, Wagtail, Jinja, etc. are kept i
...
+- `deployment.txt`: requirements for deployment, includes `base.txt` and a New Relic library which we don't install anywhere else.
```
Comment (higs4281) [inline docs/index.md:41]: minor suggestion

```suggestion
- `deployment.txt`: requirements for deployment, includes `base.txt` and a New Relic library that we don't install anywhere else.
```

File: docs/index.md (hunk 1 above)
Comment (Scotchester) [reply docs/index.md:40]: I moved this line from below, where it was still called `travis.txt`, and that question came along with it. Seems like a pertinent question that we still should resolve.

File: docs/index.md (hunk 2 above)
Comment (Scotchester) [reply docs/index.md:41]: Good catch!

File: docs/index.md (hunk 1 above)
Comment (higs4281) [reply docs/index.md:40]: Question is good; just seems out of place on the main page of our docs.
Not a blocker.

**Pull request merged**